pkg/util/proxyutil/proxyutil.go

```diff
@@ -61,3 +66,11 @@ func ClearCookieHeader(req *http.Request, keepCookiesNames []string, skipCookies
 func SetProxyResponseHeaders(header http.Header) {
 	header.Set("Content-Security-Policy", "sandbox")
 }
+
+// ApplyUserHeader Set the X-Grafana-User header if needed (and remove if not).
+func ApplyUserHeader(sendUserHeader bool, req *http.Request, user *user.SignedInUser) {
+	req.Header.Del(UserHeaderName)
+	if sendUserHeader && !user.IsAnonymous {
```

Comment:
Question: Is `user` guaranteed to be != `nil` when called from the `DataSourceProxy` and `PluginProxy` (`proxy.ctx.SignedInUser`)?

Reply:
Good point we should probably check that. I'll fix